libbeat/outputs/redis/redis.go

```diff
@@ -118,19 +124,57 @@ func makeRedis(
 	}
 
 	clients := make([]outputs.NetworkClient, len(hosts))
-	for i, host := range hosts {
-		enc, err := codec.CreateEncoder(beat, config.Codec)
+	for i, h := range hosts {
+		if parts := strings.SplitN(h, "://", 2); len(parts) != 2 {
+			h = fmt.Sprintf("%s://%s", redisScheme, h)
+		}
+
+		hostUrl, err := url.Parse(h)
+		if err != nil {
+			return outputs.Fail(err)
+		}
+
+		if hostUrl.Host == "" {
+			return outputs.Fail(fmt.Errorf("invalid redis url host %s", hostUrl.Host))
+		}
+
+		if hostUrl.Scheme != redisScheme && hostUrl.Scheme != tlsRedisScheme {
+			return outputs.Fail(fmt.Errorf("invalid redis url scheme %s", hostUrl.Scheme))
+		}
+
+		if hostUrl.Scheme == tlsRedisScheme && transp.TLS == nil {
+			transp.TLS = &transport.TLSConfig{}
+		}
+
+		var host string
+		var port int
+		if hostUrl.Port() != "" {
+			parts := strings.SplitN(hostUrl.Host, ":", 2)
```

Comment:
This might fail with IPv6. Actually `NewClient` does not need you to extract the port. The dialers in the `transp` package check if the port number is part of the hostname. If not, then they will fallback to `defaultPort`. Just use `transport.NewClient(tranp, "tcp", hustURL.Host, defaultPort)`.

Reply:
Ok I didn't know that about `transport.NewClient` so I'll remove the changes :+1: